### PR TITLE
KSQL-2720: Add section on RBAC to KSQL security docs

### PR DESCRIPTION
**Pull request by JimGalasyn**
@spena, this is my first attempt to capture what we've been discussing. The workflow for assigning roles may need to be tweaked.

### CODE REVIEW

File: docs/installation/server-config/security.rst [hunk 1]
```diff
@@ -378,9 +378,118 @@ clusters.
...
 users and groups, as well as scoping for KSQL applications.
 
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Workflow for developing a new KSQL application
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Roles and permissions to start KSQL Server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The roles mentioned in the Access to KSQL are a minimal set of roles and resources required for a user to connect to KSQL and execute KSQL commands, including transient queries. However, any command that attempts to read, write, list, and create topics require the user to have necessary permissions in Kafka and/or Schema Registry to be able to proceed with the command execution.
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: suggested rewrite

```
However, users require permissions in Kafka and/or Schema Registry to use any commands that attempt to read, write, list, and create topics.
```

File: docs/installation/server-config/security.rst [hunk 2]
```diff
@@ -378,9 +378,118 @@ clusters.
...
+
+     KsqlCluster:LITERAL:ksql-cluster
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Grant users access to resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. important::
+
+   The roles that the user has must be granted to the KSQL server user as well.
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: `The roles that the user has must also be granted to the KSQL server user.`

File: docs/installation/server-config/security.rst [hunk 3]
```diff
@@ -378,9 +378,118 @@ clusters.
...
+   The KSQL server executes the same user commands in the background and by
+   other servers connected to the same KSQL cluster.
+
+Read from a topic
+  - ``DeveloperRead`` or ``ResourceOwner`` role on the Kafka topic
+  - ``DeveloperRead`` or ``ResourceOwner`` role on the |sr| subject
+  These roles enable a user to list and read the topics from the
+  SHOW TOPIC and PRINT <topic> statements. Also, the user can CREATE streams
+  and tables from these topics, as well as execute SELECT statements on their
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: avoid `as well as` construction (replace with `and`).

**Comment by JimGalasyn** [comment]
Moving to the main docs repo.

File: docs/installation/server-config/security.rst [hunk 4]
```diff
@@ -12,6 +12,8 @@ of the other services it communicates with, like |ak-tm| and |sr|.
   :ref:`SASL for authentication <kafka_sasl_auth>`, and :ref:`authorization with ACLs <kafka_authorization>`.
 - KSQL supports :ref:`Schema Registry security features <schemaregistry_security>` such SSL for encryption
   and mutual authentication for authorization.
+- Starting in |cp| 5.3, KSQL supports role-based access control (RBAC) to
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: hyperlink to main RBAC heading? 

```
...KSQL supports :ref:`role-based access control (RBAC) <rbac-overview>` to...
```

File: docs/installation/server-config/security.rst [hunk 5]
```diff
@@ -316,6 +318,213 @@ signed by a CA trusted by the default JVM trust store.
...
+
+.. note::
+
+   Role-based access control applies only to KSQL clusters that are deployed
+   in interactive mode. RBAC isn't enabled for KSQL clusters that are deployed
+   in headless mode.
+
+Role-based access control affects these actions on a KSQL cluster:
+
+- **Add**: Add a new KSQL cluster resource by using |c3| or the Confluent CLI.
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: text-string replace for confluent CLI: `|confluent-cli|`

File: docs/installation/server-config/security.rst [hunk 6]
```diff
@@ -316,6 +318,213 @@ signed by a CA trusted by the default JVM trust store.
...
++--------------------+-------------------------+------------------------+-----------+-----------------+
+| SystemAdmin        | Yes                     | Yes                    | Yes       | Yes             |
++--------------------+-------------------------+------------------------+-----------+-----------------+
+| UserAdmin          | No                      | No                     | No        | Yes             |
++--------------------+-------------------------+------------------------+-----------+-----------------+
+
+.. important::
+
+   Principals who have the ``ClusterAdmin`` role can add and register a new
+   KSQL cluster, but we recommend against giving cluster administrators access
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: avoid using `we` ([per style guide](https://confluentinc.atlassian.net/wiki/spaces/DOC/pages/161743785/Style+Guide+for+Confluent+Documentation#StyleGuideforConfluentDocumentation-VoiceandTone))

File: docs/installation/server-config/security.rst [hunk 7]
```diff
@@ -316,6 +318,213 @@ signed by a CA trusted by the default JVM trust store.
...
+   to data. As a best practice for security, give cluster administrators only
+   explicit permissions to access specific resources within the KSQL cluster.
```
Comment (joel-hamill) [inline docs/installation/server-config/security.rst]: suggest:

```
As a best practice for security, cluster administrators should only be granted explicit...
```

**Pull request closed without merging**